libtenzir/src/tql2/ast.cpp

```diff
@@ -209,6 +253,22 @@ auto split_legacy_expression(const ast::expression& x)
         };
         TENZIR_UNREACHABLE();
       });
+      switch (y.op.inner) {
+        case ast::binary_op::gt:
+        case ast::binary_op::geq:
+          if (auto expr = optimize_now(y.left, *rel_op, y.right)) {
+            return std::pair{std::move(expr).value(), std::move(x)};
+          }
+          break;
+        case ast::binary_op::lt:
+        case ast::binary_op::leq:
+          if (auto expr = optimize_now(y.right, *rel_op, y.left)) {
+            return std::pair{std::move(expr).value(), std::move(x)};
+          }
+          break;
+        default:
+          break;
+      }
```

Comment:
I can see two issues here:

1. Potential UB due to `*rel_op` without check.
2. Swapping `right` and `left` in the `lt` and `leq` cases with otherwise identical code looks incorrect.

Reply:
1. I'll move the switch inside the check below to be explicit.
2. Thanks for the catch! I'll fix that.